lib/block_view.go

```diff
@@ -4726,135 +5590,780 @@ func (bav *UtxoView) _connectUpdateProfile(
+	}
+
+	// Get the postEntry so we can update the number of NFT copies for sale.
+	postEntry := bav.GetPostEntryForPostHash(txMeta.NFTPostHash)
+	if postEntry == nil || postEntry.isDeleted {
+		return 0, 0, nil, fmt.Errorf("_connectUpdateNFT: non-existent postEntry for NFTPostHash: %s",
+			txMeta.NFTPostHash.String())
+	}
+
+	// Verify that the updater is the owner of the NFT.
+	updaterPKID := bav.GetPKIDForPublicKey(txn.PublicKey)
+	if updaterPKID == nil || updaterPKID.isDeleted {
+		return 0, 0, nil, fmt.Errorf("_connectUpdateNFT: non-existent updaterPKID: %s",
+			PkToString(txn.PublicKey, bav.Params))
+	}
+	if !reflect.DeepEqual(prevNFTEntry.OwnerPKID, updaterPKID.PKID) {
+		return 0, 0, nil, RuleErrorUpdateNFTByNonOwner
+	}
+
+	// Sanity check that the NFT entry is correct.
+	if !reflect.DeepEqual(prevNFTEntry.NFTPostHash, txMeta.NFTPostHash) ||
+		!reflect.DeepEqual(prevNFTEntry.SerialNumber, txMeta.SerialNumber) {
+		return 0, 0, nil, fmt.Errorf("_connectUpdateNFT: prevNFTEntry %v is inconsistent with txMeta %v;"+
+			" this should never happen.", prevNFTEntry, txMeta)
+	}
+
+	// At the moment, updates can only be made if the 'IsForSale' status of the NFT is changing.
+	// As a result, you cannot change the MinBidAmountNanos of an NFT while it is for sale.
+	if prevNFTEntry.IsForSale == txMeta.IsForSale {
+		return 0, 0, nil, RuleErrorNFTUpdateMustUpdateIsForSaleStatus
+	}
+
+	// Connect basic txn to get the total input and the total output without
+	// considering the transaction metadata.
+	totalInput, totalOutput, utxoOpsForTxn, err := bav._connectBasicTransfer(
+		txn, txHash, blockHeight, verifySignatures)
+	if err != nil {
+		return 0, 0, nil, errors.Wrapf(err, "_connectUpdateNFT: ")
+	}
+
+	// Force the input to be non-zero so that we can prevent replay attacks.
+	if totalInput == 0 {
+		return 0, 0, nil, RuleErrorUpdateNFTRequiresNonZeroInput
+	}
+
+	if verifySignatures {
+		// _connectBasicTransfer has already checked that the transaction is
+		// signed by the top-level public key, which we take to be the poster's
+		// public key.
+	}
+
+	// Now we are ready to update the NFT. Two things must happen:
+	// 	(1) Update the NFT entry.
+	//  (2) If the NFT entry is being updated to "is not for sale", kill all the bids.
+	//  (3) Update the number of NFT copies for sale on the post entry.
+
+	// Create the updated NFTEntry.
+	newNFTEntry := &NFTEntry{
+		LastOwnerPKID:     prevNFTEntry.LastOwnerPKID,
+		OwnerPKID:         updaterPKID.PKID,
+		NFTPostHash:       txMeta.NFTPostHash,
+		SerialNumber:      txMeta.SerialNumber,
+		IsForSale:         txMeta.IsForSale,
+		MinBidAmountNanos: txMeta.MinBidAmountNanos,
+		// Keep the last accepted bid amount nanos from the previous entry since this
+		// value is only updated when a new bid is accepted.
+		LastAcceptedBidAmountNanos: prevNFTEntry.LastAcceptedBidAmountNanos,
+	}
+	bav._setNFTEntryMappings(newNFTEntry)
+
+	// Iterate over all the NFTBidEntries for this NFT and delete them.
```

Comment:
Don't you only want to do this if the nft is going "for sale" -> "not for sale"? I don't see a check for that here.

Maybe you just don't expect an NFT to go "not for sale" -> "for sale" and have bids on it, in which case this logic always applies. Still, feels better to have the check just in case.